lua/telescope/_extensions/file_browser/make_entry.lua

```diff
@@ -13,6 +14,9 @@ local SIZE = {
   width = 7,
   right_justify = true,
   display = function(entry)
+    if not entry.stat then
```

Comment:
I think this be would logging twice for a broken symlink as is (assuming you're both showing date and size). Let's move logging to the `stat` metatable key if it's not found :) Then we can also universally catch existence of `stat` in the loop in which it is added to displayed items similar to what I did (so no particular function has to check `stat` is available, esp. since I eventually want to allow users to hook into that.)

I'm also not sure whether invalid symlink would be the only cause of having no `stat` to begin with. Unless we're certain, let's have a more universal logging message.

Reply:
> I think this be would logging twice for a broken symlink as is (assuming you're both showing date and size).
You're right it definitely was. I have migrated the logging into a metatable.
**Note:** My experience with Lua is quite lacking so metatables are a bit of blackmagic to me. Lemme know if I can do this better.

Reply:
Metatables (MT) in brief on the example of the entry maker.

MT define a certain set of meta methods (`__index, __newindex, __call, ...`) for table `t` (`self` is also a keyword recognized by treesitter that would make sense).

The `__index = function(t, k) ... end` meta method here is key and defines how to handle access of keys not found in `t`.

The first segment
```lua
    local raw = rawget(mt, k) -- rawget is table access with bypassing meta stuff
    if raw then
      return raw
    end
```
essentially reroutes to keys defined in `mt`, i.e. `display` and `cwd` (levering closures nicely here for `display`).

```lua
    if k == "Path" then
      t.Path = Path:new(t.value)
      return t.Path
    end

    if k == "path" then
      local retpath = t.value
      if not vim.loop.fs_access(retpath, "R", nil) then
        retpath = t.value
      end
      return t.value
    end
    if k == "stat" then
      local stat = vim.loop.fs_stat(t.value)
      t.stat = vim.F.if_nil(stat, false)
      return stat
    end
```

lazily (i.e., create only on access) defining these attributes.

```lua
    return rawget(t, rawget({ value = 1 }, k))
```
telescope convention that `entry.value` and `entry[1]` will always give you the original "value" as returned by the provider (eg absolute paths from `fd` or tables with lnum col etc from lsp stuff).

Furthermore, compare this to https://github.com/nvim-telescope/telescope.nvim/blob/7bc8efd77bc99f8e0bdf49b68f11ba9de731b45c/lua/telescope/pickers.lua#L43-L44 
where we `__index` is a bit simpler and solely reroutes to the `Picker:$method` which is something you'd see more commonly.